lib/plugins/plugin/plugin.js

```diff
@@ -0,0 +1,459 @@
+              });
+            }
+            const message = 'An error occurred while installing your plugin. Please try again...';
+            this.serverless.cli.log(message);
+            return BbPromise.resolve();
+          });
+        }
+        return BbPromise.resolve();
+      });
+  }
+
+  uninstall() {
+    return BbPromise.bind(this)
+      .then(this.validate)
+      .then(this.getPlugins)
+      .then(plugins => {
+        const pluginName = this.options.name;
+        const plugin = plugins.find((item) => item.name === pluginName);
+        if (!plugin) {
+          const message = `Plugin "${pluginName}" not found. Did you spell it correct?`;
+          this.serverless.cli.log(message);
+        } else if (!this.checkPluginExists()) {
+          const message = `Plugin "${pluginName}" has been already uninstalled.`;
+          this.serverless.cli.log(message);
+        } else {
+          // uninstall the package through npm
+          this.serverless.cli
+            .log(`Uninstalling plugin "${pluginName}" (this might take a few seconds...)`);
+
+          return BbPromise.bind(this)
+          .then(this.uninstallPeerDependencies)
+          .then(this.pluginUninstall)
+          .then(pluginStillAvailable => {
+            if (!pluginStillAvailable) {
+              return this.removePluginFromServerlessFile()
+              .then(() => {
+                this.serverless.cli.log(`Successfully uninstalled "${pluginName}"`);
+              });
+            }
+            const message = 'An error occurred while uninstalling your plugin. Please try again...';
+            this.serverless.cli.log(message);
+            return BbPromise.resolve();
+          });
+        }
+        return BbPromise.resolve();
+      });
+  }
+
+  list() {
+    return BbPromise.bind(this)
+      .then(this.getPlugins)
+      .then((plugins) => this.display(plugins));
+  }
+
+  search() {
+    return BbPromise.bind(this)
+      .then(this.getPlugins)
+      .then((plugins) => {
+        // filter out plugins which match the query
+        const regex = new RegExp(this.options.query);
+
+        const filteredPlugins = plugins.filter((plugin) =>
+          (plugin.name.match(regex) || plugin.description.match(regex))
+        );
+
+        // print a message with the search result
+        const pluginCount = filteredPlugins.length;
+        const query = this.options.query;
+        const message = `${pluginCount} plugin(s) found for your search query "${query}"\n`;
+        this.serverless.cli.consoleLog(chalk.yellow(message));
+
+        return filteredPlugins;
+      })
+      .then((plugins) => {
+        this.display(plugins);
+      });
+  }
+
+  // helper methods
+  validate() {
+    if (!this.serverless.config.servicePath) {
+      throw new this.serverless.classes
+        .Error('This command can only be run inside a service directory');
+    }
+
+    return BbPromise.resolve();
+  }
+
+  getServerlessFilePath() {
```

Comment:
This could be replace with this util function: https://github.com/serverless/serverless/blob/master/lib/utils/getServerlessConfigFile.js